### PR TITLE
Automatically restart the worker when the catalog changes

### PR DESCRIPTION
**Pull request by jimyag**

### CODE REVIEW

File: charts/trino/templates/deployment-worker.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ spec:
     metadata:
       annotations:
         checksum/worker-config: {{ include (print $.Template.BasePath "/configmap-worker.yaml") . | sha256sum }}
+        checksum/catalog-config: {{ include (print $.Template.BasePath "/configmap-catalog.yaml") . | sha256sum }}
```
Comment (nineinchnick) [inline charts/trino/templates/deployment-worker.yaml:25]: Nit: let's keep them in the same order as in `deployment-coordinator.yaml`
```suggestion
        checksum/catalog-config: {{ include (print $.Template.BasePath "/configmap-catalog.yaml") . | sha256sum }}
        checksum/worker-config: {{ include (print $.Template.BasePath "/configmap-worker.yaml") . | sha256sum }}
```

**Review by nineinchnick — APPROVED** [review]

**Review by nineinchnick — APPROVED** [review]
Please squash both commits. I can merge this once the CLA clears out.

**Comment by nineinchnick** [comment]
@cla-bot check

**Review by mosabua — COMMENTED** [review]
Do we really want to do this given that we have dynamic catalogs that work with these changes gracefully. Does this approach actually work for running queries or so? Also .. I am actually wondering if the same applies to worker-config ..

**Comment by jimyag** [comment]
@cla-bot check

**Comment by jimyag** [comment]
@cla-bot check

**Comment by jimyag** [comment]
@cla-bot check

**Pull request merged**